app/services/stats/base_monthly_performance_report.rb

```diff
@@ -64,7 +98,40 @@ def to_csv
       end
     end
 
-    private
+    # def required_to_be_job
+    #   return false 
+    # end
+
+    # This function is only when the report is done via ETL (tasks)
+    def report_details(report)
+      redis = Redis.new
+      data_collector = []
+      report.job_ids.each do |job_id|
+        if redis.exists(job_id)
+          data_collector << redis.get(job_id)
+        end
+      end
+      if data_collector.count == report.job_ids.count
```

Comment:
I think this method also might benefit from being split up a bit into smaller functions